geode-core/src/main/java/org/apache/geode/cache/client/internal/pooling/ConnectionManagerImpl.java

```diff
@@ -338,6 +335,10 @@ public PooledConnection borrowConnection(ServerLocation server, long acquireTime
               }
             }
           }
+          if (null != connection) {
+            return connection;
+          }
+          throw new ServerConnectivityException(BORROW_CONN_ERROR_MSG + server);
         }
 
         if (!onlyUseExistingCnx) {
```

Comment:
Sorry, if i missed this in earlier comments. The "onlyUseExistingCnx" indicates no new connection to be created, by the time this gets checked, a create connection attempt will be made if "tryCreate" is true (above logic). 
It seems right to keep the condition check as If(onlyUseExistingCnx) { throw new AllConnectionsInUseException();} if there are no connection to this server available, the exception may be handled differently based on the caller.

Reply:
I introduced logic simular to borrowConnection, because if connectins in pool are not up to maximum, why we could not create new connection. And if "onlyUseExistingCnx" is set to false, we will create new connection, even if we exceed maximum.
But, I can return to previous implementation.

Reply:
Agree with your thinking in relation to max-connection; but this arguments seems to enforce that only existing connection needs to be used.